samples/simple_displace.rb

```diff
@@ -1,17 +1,100 @@
+    @app = app
+    @art = art
+    @elements = elements
+    @position = {}
+    @width = 50
+    @margin = 10
+    @top = 4
+  end
+
+  attr_reader :art, :elements, :width, :margin, :top
+
+  def oval
+    @app.oval(*position(:oval), width)
+  end
+
+  def rect
+    @app.rect(*position(:rect), width)
+  end
+
+  def image
+    @app.image File.expand_path(File.join(__FILE__, "../avatar.png")), left: left(:image), top: top
+  end
+
+  def edit_box
+    @app.edit_box "Edit me", left: left(:edit_box), top: top, height: 50, width: width
+  end
+
+  def offset(element)
+    art.index(element.to_sym) || elements.index(element.to_sym)
+  end
+
+  def left(element)
+    (width + margin) * offset(element)
+  end
+
+  def position(element)
+    [left(element), top]
+  end
+
+  def remember(name, element)
+    @app.info "Remembering #{name} at [#{element.left}, #{element.top}]"
+    @position[name] = [element.left, element.top]
+  end
+
+  def recall(name)
+    position = @position.fetch(name, "...oops, can't find it...[0, 0]")
+    @app.info "Recalling #{name} was at #{position}"
+    @position.fetch(name, [0, 0])
+  end
+end
+
 Shoes.app do
-  # Shoes 3 doesn't support button.text
-  position_report = ->(button, button_text) { @position_report.replace "#{button_text}: (#{button.left}, #{button.top})" }
-  flow margin: 12 do
-    # Set up three buttons
-    button "One", click: proc { |button| position_report.call(button, "One") }
-    @two = button "Two", click: proc { |button| position_report.call(button, "Two") }
-    button "Three", click: proc { |button| position_report.call(button, "Three") }
-    @position_report = para ''
-    # Bounce the second button
-    animate do |i|
-      @two.displace(0, (Math.sin(i) * 6).to_i)
+  @art = [:oval, :rect]
+  @elements = [:image, :edit_box]
+  @objects = {}
+  @layout = Layout.new self, @art, @elements
+  @create_object = lambda do |name|
+    info "#{name} starts at #{@layout.position(name)}"
+    @objects[name] = @layout.send(name)
+    @layout.remember name, @objects[name]
+  end
+
+  stack do
+    title "Art"
+    flow height: 100 do
+      @art.each(&@create_object)
+    end
+
+    title "Elements"
+    flow height: 100 do
+      @elements.each(&@create_object)
+    end
+
+    flow do
+      x = 400
+      colors = [seagreen, salmon, papayawhip, goldenrod]
+      @art.concat(@elements).each_with_index do |name, i|
+        stack width: 100, margin: 4 do
+          background colors[i]
+          tagline name.to_s
+          button "displace" do
+            @layout.remember(name, @objects[name])
+            position = [x - @layout.left(name), @layout.top]
+            info "#{name} displaces to #{position}"
+            @objects[name].displace(*position)
+          end
+          button "replace" do
+            position = @layout.recall(name)
+            info "#{name} replaces to #{position}"
+            @objects[name].displace(*position)
+          end
+        end
+      end
```

Comment:
this... uhhh exploded in size?

Reply:
Odd. I suspect I removed the original simple_displace for some reason, then renamed over it later on. Will review the commits to make sure that was intentional.

Reply:
Confirmed... what's now `simple_displace.rb` was `good_displace.rb` before, and the old `simple_displace.rb` was a bouncing button of which we have another similar copy. Relevant commits:
* https://github.com/shoes/shoes4/pull/1556/commits/df8e791d50b7308ec54347c32b4ff86561fc5a51
* https://github.com/shoes/shoes4/pull/1556/commits/59866b3c970fcc7d79a80adfed2bb8af059123ed

Reply:
thanks!